src/App.tsx

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import { useState, useEffect } from 'react';
+import { HashRouter as Router } from 'react-router-dom';
+
+import { AddTodoForm } from './components/AddTodoForm';
+import { TodoList } from './components/TodoList';
+import { TodosFilter } from './components/TodosFilter';
+import { Todo } from './types/types';
+
+export const App: React.FC = () => {
+  const [inputText, setInputText] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState('all');
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+
+  const filterHandler = () => {
+    switch (status) {
+      case 'completed':
+        setFilteredTodos(todos.filter(todo => todo.completed));
+        break;
+      case 'active':
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+        break;
+      default:
+        setFilteredTodos(todos);
+    }
+  };
+
+  const saveLocalTodos = () => {
+    localStorage.setItem('todos', JSON.stringify(todos));
+  };
+
+  const getLocalTodos = () => {
+    if (localStorage.getItem('todos') === null) {
+      localStorage.setItem('todos', JSON.stringify([]));
+    } else {
+      setTodos(JSON.parse(localStorage.getItem('todos') || ''));
+    }
+  };
```

Comment:
better to craete separate folder for hooks

Reply:
But then there will be a nessesary to import `todos`, `setTodos` from `App`, crossimporting is not very good practice

Reply:
> But then there will be a nessesary to import todos, setTodos

You can pass them as a param to the function, so move it out of your component

Reply:
got it